satpy/composites/__init__.py

```diff
@@ -1677,23 +1677,74 @@ def __call__(self, *args, **kwargs):
 
 
 class BackgroundCompositor(GenericCompositor):
-    """A compositor that overlays one composite on top of another."""
+    """A compositor that overlays one composite on top of another.
 
-    def __call__(self, projectables, *args, **kwargs):
+    Beside foreground and background, a third optional dataset could be passed
+    to the compositor to use it for masking. If this dataset contains
+    more than one band, only the first band will be used. This is useful when you
+    reproject a specific local-area image (e.g. a geostationary satellite view)
+    to global extent, and put it on a global background (e.g. NASA's Black Marble)
+    while making other areas of the world transparent, only keeping the local one.
+
+    To use this function in a YAML configuration file, add the third dataset
+    as ``optional_prerequisites``:
+
+    .. code-block:: yaml
+
+      night_cloud_alpha_2000_with_background:
+        compositor: !!python/name:satpy.composites.BackgroundCompositor
+        prerequisites:
+          - name: night_cloud_alpha_2000
+          - name: static_night
+        optional_prerequisites:
+          - name: IR105
+
+    """
+    def __init__(self, name, mask_value=None, **kwargs):  # noqa: D417
+        """Collect custom configuration values.
+
+        Args:
+            mask_value (float / None): All the pixels on the stacked image where the values
+                                       of the third dataset that are equal to this will be
+                                       masked out.
+                                       If not set by the user, it will be ``np.nan``.
+                                       This argument could be helpful when you try to use a
+                                       local image(StaticImageCompositor) for masking.
```

Comment:
Reworded this, but also types should be separated by I think a vertical bar `|` or maybe a comma in google style docstrings. Let's see what the readthedocs job says.

```suggestion
            mask_value (float | None): Value of the third dataset used to generate
                the mask for the stacked image. Defaults to ``np.nan``. This is
                useful when wanting additional masking for a static image
                (e.g. :class:`StaticImageCompositor`).
```